ci/ci

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import platform
+import argparse
+import subprocess
+import fileinput
+import shlex
+import time
+
+pbs_dirname = ''
+
+
+def log_error(str_to_print):
+    t = time.localtime()
+    current_time = time.strftime("%H:%M:%S", t)
+    print('\033[91m' + current_time + " ---> " + str_to_print + '\033[0m')
+
+
+def log_info(str_to_print):
+    t = time.localtime()
+    current_time = time.strftime("%H:%M:%S", t)
+    print('\033[1m' + current_time + " ---> " + str_to_print + '\033[0m')
+
+def run_cmd(cmd, return_output=False):
+    cmd = shlex.split(cmd)
+    try:
+        a = subprocess.Popen(cmd,stdout=subprocess.PIPE)
+        out,err = a.communicate()
+        if a.returncode != 0:
+            log_error("command failed")
+            log_error(str(err))  
+        else:
+            if return_output:
+                return str(out)
+    except Exception as e:
+        log_error("The command failed.")
+        log_error(e)
+
+
+def run_docker_cmd(cmd):
+    try:
+        docker_cmd = shlex.split(cmd)
+        a = subprocess.Popen(docker_cmd)
+        a.communicate()
+        if a.returncode != 0:
+            log_error(
+                "docker cmd returned with non zero exit code, dropping you into container")
+            docker_cmd = shlex.split(
+                "docker-compose exec pbs-build bash -c \'cd /pbspro && /bin/bash\'")
+            subprocess.run(docker_cmd)
+    except Exception as e:
+        log_error("Failed\n:")
+        log_error(e)
+
+
+def write_to_file(file_path, value):
+    with open(file_path, "w+") as f:
+        f.write(value)
+
+
+def read_from_file(file_path):
+    if not os.path.isfile(file_path):
+        open(file_path, 'a').close()
+    with open(file_path, 'r+') as f:
+        val = f.read()
+    return val
+
+
+def commit_docker_image():
+    env_dir = os.path.join(pbs_dirname, ".env")
+    image_name = read_from_file(env_dir)
+    image_name = image_name.split('=')[1]
+    if image_name.find('_ci_pbs') == -1:
+        image_name += '_ci_pbs'
+        image_name = image_name.replace(':', '_')
+        try:
+            build_id = 'docker-compose ps -q pbs-build'
+            build_id = run_cmd(build_id,True)
+            build_id = build_id.split("'")[1]
+            build_id = build_id[:12]
```

Comment:
explain the magic number 12 here.

Reply:
It shortens the container id returned by **docker-compose** to the one displayed by **docker** command, as docker-compose command will return full container id.
It will be then be displayed as something below 
`14:21:46 ---> docker commit 50315b45f569 centos_7_ci_pbs:latest`
It's helpful to user as the container id is similar to one shown by command "docker ps"
```
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES
50315b45f569        centos:7            "src/docker-entrypoi…"   2 minutes ago       Up 2 minutes                            ci_pbs-build_1
```

Reply:
okay. thank you for the explanation. we need to add a comment in the code to explain why are we using 12 instead of any other number.